python/lancedb/embeddings/gte.py

```diff
@@ -0,0 +1,127 @@
+#  Copyright (c) 2023. LanceDB Developers
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from typing import List, Union
+
+import numpy as np
+
+from .base import TextEmbeddingFunction
+from .registry import register
+from .utils import weak_lru
+
+
+
+@register("gte-text")
+class GteEmbeddings(TextEmbeddingFunction):
+    """
+    An embedding function that uses GTE-LARGE MLX format (for Apple silicon devices only)
+    as well as the standard cpu/gpu version from huggingface - https://huggingface.co/thenlper/gte-large    
+    
+    Parameters
+    ----------
+    name: str, default "thenlper/gte-large"
+        The name of the model to use.
+    device: str, default "cpu"
+        Sets the device type for the model.
+    normalize: str, default "True"
+        Controls normalize param in encode function for the transformer
+    flag: str, default "gte-large"
+        Controls which model to use. Options are "gte-large" or "mlx"
+
```

Comment:
what is the difference bw name and flag if both control the model

Reply:
name is for the sentence-transformer module to get the correct model, flag controls the switch to mlx format.
I can change the name for the flag for the standard model if needed , let me know.

Reply:
ok let's call it `mlx: boolean =False`